From: Andreas Kloeckner <inform@tiker.net>
Date: Tue, 25 Dec 2012 19:36:42 +0000 (+0100)
Subject: Update tagpy for TagLib 1.7 compatibility. (patch by unknown Gentoo user, via Lars... 
X-Git-Tag: v2013.1~6
X-Git-Url: http://git.tiker.net/tagpy.git/commitdiff_plain/d0759c0ec7d24ae8d1af395a1032521eb00b4d1f?hp=c2801c31493f77d7c33d20fa20f0b298b0f322ef

Update tagpy for TagLib 1.7 compatibility. (patch by unknown Gentoo user, via Lars Wendler)
---

diff --git a/src/wrapper/id3.cpp b/src/wrapper/id3.cpp
index a76f7ef..5c08ae1 100644
--- a/src/wrapper/id3.cpp
+++ b/src/wrapper/id3.cpp
@@ -79,6 +79,10 @@ namespace
   MF_OL(setVolumeAdjustment, 1, 2);
   MF_OL(setPeakVolume, 1, 2);
 
+  #if (TAGPY_TAGLIB_HEX_VERSION >= 0x10700)
+    MF_OL(render, 0, 1)
+  #endif
+
   // -------------------------------------------------------------
   // MPEG
   // -------------------------------------------------------------
@@ -212,7 +216,11 @@ void exposeID3()
       .DEF_SIMPLE_METHOD(removeFrame)
       .DEF_SIMPLE_METHOD(removeFrames)
       
-      .DEF_SIMPLE_METHOD(render)
+      #if (TAGPY_TAGLIB_HEX_VERSION >= 0x10700)
+        .DEF_OVERLOADED_METHOD(render, ByteVector (cl::*)(int) const)
+      #else
+        .DEF_SIMPLE_METHOD(render)
+      #endif
       ;
   }
 
